### ISSUE TITLE
Positive feedback on a proof

### ISSUE THREAD

**Issue by lovettchris** [leanprover/lean4#980]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your issue isn't already [filed](https://github.com/leanprover/lean4/issues).
    * Reduced the issue to a self-contained, reproducible test case.

### Description

When I write a valid proof like:

```lean
theorem t1 {p q : Prop} (hp : p) (hq : q) : p := hp
```

and put the cursor at the end of the line the output from LSP is boring.
![image](https://user-images.githubusercontent.com/18707114/151646525-840d530d-e0f4-4342-b430-cd56c82a966d.png)

I think it would be fun and "more addictive" if Lean gave some sort of nice output here saying something like "theorem t1 is valid!" or something.

### Steps to Reproduce

See above

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions

You can get this information from copy and pasting the output of `lean --version`,
please include the OS and what version of the OS you're running.

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by Kha** [comment]
Basically all nontrivial Lean proofs are written in tactic mode where we do have a nice emoji.

**Comment by lovettchris** [comment]
Great so can you add a nice emoji on this one too?

**Comment by leodemoura** [comment]
@lovettchris The emojis are added by the VS Code plugin.

**Comment by lovettchris** [comment]
Oh, you are right, I just found ` return <>Goals accomplished 🎉</>` and ok, so now I need to investigate why it sometimes doesn't work ...

**Comment by lovettchris** [comment]
@leodemoura, Ok, I investigated and I think I found an LSP issue.  The problem is a theorem with tactic state like this 
```lean
theorem test (p q : Prop) (hp : p) (hq : q) : p ∧ q ∧ p :=
  by apply And.intro
     exact hp
     apply And.intro
     exact hq
     exact hp
```

returns this when the cursor is at the end of the proof:

```json
{"result":{"goals":[]},"jsonrpc":"2.0","id":28}
```
and the zero length goals array tells the vscode plugin to show the nice `Goals accomplished 🎉` feedback.

But a theorem like this:
```
theorem and_com (h : p ∧ q) : q ∧ p := ⟨h.right, h.left⟩
```
returns this when the cursor is at the end of the proof:
```json
{"result":null,"jsonrpc":"2.0","id":39}
```
Notice there is no Goals array at all so the vscode plug has no idea this is a proof.  Could just be anything, so the entire "Tactic state" block is hidden, including the goals accomplished message.

So can the LSP be changed to return a zero length set of Goals in this second case?
